### PR TITLE
Add `comparison_rate_type` to ShipEngine rates requests

### PR DESCRIPTION
**Pull request by pelargir**
When `comparison_rate_type` is set to "retail" ShipEngine returns retail rates (instead of commercial rates) from the rates API endpoint. Note that this only works for UPS/USPS carriers.

### CODE REVIEW

File: lib/friendly_shipping/services/ship_engine/rates_options.rb [hunk 1]
```diff
@@ -9,26 +9,40 @@ class ShipEngine
       #
       # @attribute carriers [Array<FriendlyShipping::Carrier>] a list of the carriers we want to get IDs from.
       # @attribute service_code [String] The service code we want to get rates for.
+      # @attribute ship_date [Time] The date we want to ship on.
```
Comment (davecandlescience) [inline lib/friendly_shipping/services/ship_engine/rates_options.rb:12]: nit: is it a date or a time?

**Comment by mamhoff** [comment]
I'm just wondering: Do all of our services support some kind of distinction between retail, commercial and negotiated rates? Should we adopt a convention around that?

**Comment by davecandlescience** [comment]
When would a rates request ever be missing an origin or destination address? That seems an odd case to test that it gets passed through to the api correctly.

**Comment by pelargir** [comment]
> When would a rates request ever be missing an origin or destination address? That seems an odd case to test that it gets passed through to the api correctly.

It was more about keeping the test short but I agree that it could be confusing. I'll fix.

**Comment by pelargir** [comment]
> I'm just wondering: Do all of our services support some kind of distinction between retail, commercial and negotiated rates? Should we adopt a convention around that?

They do, but it's kind of a mess as far as how it works. With USPS, for example, you can either pass a service name ("PRIORITY COMMERCIAL" for example) or you can pass "ONLINE" which will return both retail _and_ commercial rates for all available services. Whereas, UPS doesn't really have a concept of "commercial" rates, only "negotiated" rates which can be returned in the API response by including the `NegotiatedRatesIndicator` in the request.

I'm not sure what kind of convention we could adopt given that not every carrier has the retail/commercial distinction.

File: lib/friendly_shipping/services/ship_engine/rates_options.rb (hunk 1 above)
Comment (pelargir) [reply lib/friendly_shipping/services/ship_engine/rates_options.rb:12]: Good point. I'll fix.

Comment (pelargir) [reply lib/friendly_shipping/services/ship_engine/rates_options.rb:12]: (But in a separate PR since the change is out of scope for this one.)

**Review by davecandlescience — APPROVED** [review]
LGTM

**Pull request merged**